chipsec/logger.py

```diff
@@ -381,87 +401,27 @@ def logger():
 # Hex dump functions
 ##################################################################################
 
+def chunk (array, width = 8):
+    return [ array[i: i + width] for i in range(0, len(array), width) ]
+
 def dump_buffer( arr, length = 8 ):
     """Dumps the buffer."""
-    tmp=[]
-    tmp_str=[]
-    i=1
-    for c in arr:
-        tmp+=["{:02x} ".format(ord(c))]
-        if c in string.whitespace or c not in string.printable:
-            ch = " "
-        else:
-            ch = ord(c)
-        tmp_str+=["%c"%ch]
-        if i%length==0:
-            tmp+=["| "]
-            tmp+=tmp_str
-            tmp+=["\n"]
-            tmp_str=[]
-        i+=1
-    if 0 != len(arr)%length:
-        tmp+=[ (length - len(arr)%length) * 3*" " ]
-        tmp+=["| "]
-        tmp+=tmp_str
-        tmp+=["\n"]
-    return "".join(tmp)
+
+    result = ""
+    pieces = chunk(arr, length)
+    for piece in pieces:
```

Comment:
Try this:
```
if isinstance(arr,str):
    left = "".join(["{0>2x} ".format(ord(c)) for c in piece])
    right = "".join("{["{:s} ".format(c) if c in string.printable else '. ' for c in piece ])
    line = "{: <{}} | {:s}\n".format(left,length*3,right)
else:
    line 413 - 415
```

Reply:
This can work for time being. I started exploring something based on bytearray that works for the bios_kbrd_buffer case on both Py2 and Py3.

but then started looking into the acpi/UEFI failure and then I found that there is a lot of code doing encode/decode operations to read binary blobs and print them out to screen. I think we probably should think refactoring and unify the code in a compatible way. I'll post a follow up based on your proposal.

Reply:
This is already taken care of in #878, no need to do anymore